src/Mod/Draft/draftgeoutils/geometry.py

```diff
@@ -219,6 +219,72 @@ def get_spline_normal(edge, tol=-1):
         return None
 
 
+def get_shape_normal(shape):
+    """Find the normal of a shape or list of points or colinear edges, if possible."""
+
+    # New function based on get_normal() drafted by @Roy_043  
+    # in discussion https://forum.freecad.org/viewtopic.php?p=717862#p717862 
+    #
+    # The normal would not be affected  by the 3D view direction and flipped as
+    # get_normal would be. Placement of the Shape is taken into account in this
+    # new function instead :
+    # - Normal of a planar shape is usually bi-directional, this function return
+    #   the one 'in the direction' of the shape's placement 'normal' (z-direction).
+    #   As reference, shape.findPlane() favour positive axes, get_normal()
+    #   favour the 'view direction' (obtained by getViewDirection() ).
+    # - Even when the Shape is an edge or colinear edges, its Placement is taken
+    #   into account, and this function return one 'in the direction' of the
+    #   shape's placement 'normal' (z-direction).
+    #   https://forum.freecad.org/viewtopic.php?p=715850#p715850
+    # The normal direction of a Draft wire (co-planar) or arc used to depends on
+    # the way the shape is constructed i.e. by vertexes in clockwise or
+    # anti-clockwise.  This cryptic behaviour no longer matters.
+
+    # for points
+    if isinstance(shape, (list, tuple)):
+        if len(shape) <= 2:
+            return None
+        else:
+            shape = Part.makePolygon(shape)
```

Comment:
This will fail in a number of cases, no? For ex. if the points are not coplanar. Better either put more tests, or place this inside a try statement

Reply:
Indeed, could improve, in fact I just more or less copy these from get_normal()  :D